### ISSUE TITLE
InternalServerError when trying to configure new build agent for TFS 2017.2

### ISSUE THREAD

**Issue by NenoLoje** [Microsoft/vsts-agent#1203]
## What's not working?
When running `config.cmd`, it fails with:

> Microsoft.VisualStudio.Services.Common.VssServiceException: An internal error occurred.

On the Agent Pools page in TFS, the new build agent does get created, but the configuration fails before offering to create the windows service entry.

## From the agent logs:
 
```
[2017-09-11 12:03:17Z INFO Terminal] WRITE LINE: Connecting to the server.
[2017-09-11 12:03:17Z INFO CommandSettings] Flag 'replace': 'False'
[2017-09-11 12:03:17Z INFO CommandSettings] Flag 'unattended': 'False'
[2017-09-11 12:03:17Z INFO PromptManager] ReadValue
[2017-09-11 12:03:17Z INFO Terminal] WRITE: Enter replace? (Y/N) (press enter for N) > 
[2017-09-11 12:03:17Z INFO Terminal] READ LINE
[2017-09-11 12:03:18Z INFO Terminal] Read value: 'y'
[2017-09-11 12:03:19Z ERR  VisualStudioServices] PUT request to http://apktst02:8080/tfs/_apis/distributedtask/pools/1/agents/9 failed. HTTP Status: InternalServerError
[2017-09-11 12:03:19Z INFO CommandSettings] Flag 'unattended': 'False'
[2017-09-11 12:03:19Z ERR  Terminal] WRITE ERROR (exception):
[2017-09-11 12:03:19Z ERR  Terminal] Microsoft.VisualStudio.Services.Common.VssServiceException: An internal error occurred.

   at Microsoft.VisualStudio.Services.WebApi.VssHttpClientBase.<HandleResponseAsync>d__51.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.VisualStudio.Services.WebApi.VssHttpClientBase.<SendAsync>d__49.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.VisualStudio.Services.WebApi.VssHttpClientBase.<SendAsync>d__46`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.VisualStudio.Services.WebApi.VssHttpClientBase.<SendAsync>d__27`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Services.Agent.Listener.Configuration.ConfigurationManager.<ConfigureAsync>d__6.MoveNext()
[2017-09-11 12:03:19Z ERR  Terminal] WRITE ERROR: Failed to replace the agent.  Try again or ctrl-c to quit
[2017-09-11 12:03:19Z INFO CommandSettings] Arg 'agent': ''
[2017-09-11 12:03:19Z INFO CommandSettings] Flag 'unattended': 'False'
[2017-09-11 12:03:19Z INFO PromptManager] ReadValue
[2017-09-11 12:03:19Z INFO Terminal] WRITE: Enter agent name (press enter for APKQA01) > 
[2017-09-11 12:03:19Z INFO Terminal] READ LINE
[2017-09-11 12:04:01Z INFO Terminal] WRITE LINE: Exiting...
```

## Agent versions tried:
I tried the following versions:
- v2.122.1 (Latest Prerelease)
- v2.120.2 (Latest release)
- v2.117.2 (TFS2017 QU2)

## OS of the machine running the agent:
Windows Server 2008 R2 SP1

## TFS version:
TFS 2017 Update 2 (2017.2)

## Things we've already checked/tried:
- Checked the Windows event log on the TFS Application Tier. No errors or warnings from TFS.
- Using a different account.
- Adding the account to an appropriate *Access Level* explicitly.
- Removing the permissions from the account (to check that a message gets shown that the account is missing the *Manage Agent Pool* permission) --> worked as expected.

**Comment by stephenmichaelf** [comment]
@NenoLoje Could you email the diag logs to:

stfrance at microsoft.com
tihuang at microsoft.com

Diag logs can contain sensitive information so we recommend not posting them publicly.

Thanks! Then we can check into this issue.

**Comment by NenoLoje** [comment]
Here's a solution/workaround to fix the issue:

- We discovered that on the TFS Application Tier (**not** the build machine!) the windows service `CNG isolation service` was not running.
- Starting it (manually) fixed the issue and we could configure new build agents successfully.
- To make sure the issue does not accor again, we changed the **startup type** of the `CNG isolation service` to **Automatic**.

Big thanks to @stephenmichaelf to help us isolate and mitigrate the issue!

--Neno
